### DISCUSSION TITLE
كلمة شيفرة و استعمالاتها المتعددة

### DISCUSSION THREAD

**Discussion by boussadjra** [vuejs-translations/docs-ar#75] [مصطلحات]
من خلال تعريبي للعديد من الصفحات مررت بعدة مصطلحات و التي ترجمتها إلى مصطلح شيفرة/ شيفرة برمجية و هذه أهمها:
- كلمة code بشكل عام تترجم إلى شيفرة
- كلمة logic تترجم إلى شيفرة برمجية، لأنه غالبا يقصد بها الجانب البرمجي من التطبيق/المكون لتمييزها عن شيفرة المحتوى و التنسيق (HTML/CSS)
- كلمة concern، هذه الكلمة تترجم بشكل بديهي إلى اهتمام و لكنها لا تفي بالمعنى، و غالبا يقصد بها نوع الشيفرة و وظيفتها. و إذا كنا داخل نظام فيقصد بها طبيعة و نوع كل جزء منه.

مرحبا بآرائكم.

**Comment by Abdelaziz18003** [comment]
جميل، بالنسبة للمثال الثالث، أرى أن كلمة "وظيفة" أدق وأكثر تعبيرا عن معنى "concern"، وهذه ترجمة بعض الأمثلة الشائعة: 
- mixing of different concerns = خلط الوظائف المختلفة
- separation of concerns = الفصل بين الوظائف

**Comment by boussadjra** [comment]
نعم، كلمة الوظائف تفي بالغرض إلى حد بعيد، من خلال قولي "نوع الشيفرة و وظيفتها" كنت قد اقتربت من المعنى، و بالتالي من الجيد تجنب استعمال كلمة شيفرة لأنها تكررت بأكثر من معنى و هذا قد يؤدي إلى الالتباس.
